### DISCUSSION TITLE
Modifying the SOX digital filters

### DISCUSSION THREAD

**Discussion by MGU-GIT** [mikebrady/shairport-sync#1453] [General]
Hi,

Is it possible to change the digital filters in SOX using a configuration file or any other interface in Shairport-sync? It would be nice to have the opportunity to experiment and try out different (slower) filters. This is an example of what I am trying to achieve.

http://archimago.blogspot.com/2018/01/musings-more-fun-with-digital-filters.html

**Comment by MGU-GIT** [comment]
By further looking at the code, I see now that this can not be done in SOX, but needs to be done in code. Currently the interpolation is set by the _soxr_oneshot_ call which uses _SOXR_LQ_ (low quality). Is it some way of modifying this and set other qualities? Can anybody provide some input to what the differences between low and high quality is? From my understanding, the only thing which can be adjusted is from "basic" or "sox". What is then the difference between basic and sox when low quality is chosen?

**Comment by mikebrady** [comment]
Thanks. The SoX settings are default ones. One of the important considerations is how long the interpolation takes, especially on lower-powered CPUs. I imagine it would be easy to experiment simply by editing the code for configuring calls to `soxr_oneshot`.

**Reply by mikebrady** [reply]
> Hi,
> 
> Is it possible to change the digital filters in SOX using a configuration file or any other interface in Shairport-sync? It would be nice to have the opportunity to experiment and try out different (slower) filters. This is an example of what I am trying to achieve.
> 
> http://archimago.blogspot.com/2018/01/musings-more-fun-with-digital-filters.html

That's an interesting resource. But please note that in Shairport Sync, SoX is being use to resample blocks of PCM frames so as to add or remove one to do synchronisation -- it's not being used as a filter as such.

**Reply by MGU-GIT** [reply]
Thank you! If I understand you correct, this is very good news. If SOX is used in such a way that it does not apply any lowpass filter when resampling (like it usually does), then I will not have any problems with ringing filters either. That removes my initial question and concern. 

Thank you for the great work you are doing!
